### PR TITLE
chore(PPDSC-2625): fail build on cms issues

### PR DESCRIPTION
**Pull request by mstuartf**
PPDSC-2625
**What**

1. Background - CMS issues failing silently
2. What did you do - fail build for CMS connection or content issues 
3. What does the reviewers should expect
- Build should fail if CMS connection fails or content invalid
- TS support for page content objects

Spike writeup: https://nidigitalsolutions.jira.com/wiki/spaces/NPP/pages/4264067079/PPDSC-2625+Handle+CMS+connection+and+content+issues

Failure examples:
Screenshot 2023-01-13 at 12 15 58
Screenshot 2023-01-13 at 12 22 35
Screenshot 2023-01-13 at 12 34 03

**I have done:**
- [ ] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [ ] Updated relevant documentation

**I have tested manually:**
- [ ] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [ ] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

### CODE REVIEW

**Comment by pp-serviceaccount** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2625-cms-errors
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2625-cms-errors/storybook

File: site/utils/google-sheet/utils.ts [hunk 1]
```diff
@@ -11,7 +17,50 @@ export const formatSheetData = (data: string[][] | undefined | null) => {
...
-) => Object.entries(content).filter(entry => entry[0].startsWith(listKey));
+export const parseCMSResponse = <T extends CMSProps>(
+  cmsResponse: CMSResponse,
+  {
+    required,
+    optional = {},
+    dynamic = {},
+  }: {
+    required: Record<string, string>;
+    optional?: Record<string, string>;
```
Comment (Xin00163) [inline site/utils/google-sheet/utils.ts:28]: This is an interesting one. I know currently we are being explicit and any newly added keys will be invalid unless we add them in the code as well. I am just wondering if we were to add new keys, how do we decide which are optional? It seems to me that they are either required or dynamic.

Comment (mstuartf) [reply site/utils/google-sheet/utils.ts:28]: Yeah, that's a fair point. I added it to cover all cases but you're right that all keys used right now are required or dynamic. I can remove the optional stuff and we can add it back if we need it?

Comment (Xin00163) [reply site/utils/google-sheet/utils.ts:28]: Yeah that sounds good to me

**Review by mutebg — APPROVED** [review]

**Review by mutebg — APPROVED** [review]

**Pull request merged**